### PR TITLE
Worm/br support debug trace block i1523

### PR DESCRIPTION
**Pull request by lcmmhcc**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by ItsFunny — APPROVED** [review]

**Review by ItsFunny — APPROVED** [review]

**Review by ItsFunny — DISMISSED** [review]

File: app/rpc/namespaces/debug/api.go [hunk 1]
```diff
@@ -20,6 +23,7 @@ type PublicDebugAPI struct {
 	clientCtx clientcontext.CLIContext
 	logger    log.Logger
 	backend   backend.Backend
+	Metrics   map[string]*monitor.RpcMetrics
```
Comment (Ywmet) [inline app/rpc/namespaces/debug/api.go:26]: is this thead safe

File: app/rpc/namespaces/debug/api.go [hunk 2]
```diff
@@ -72,3 +77,70 @@ func (api *PublicDebugAPI) TraceTransaction(txHash common.Hash, config evmtypes.
...
 	return decodedResult, nil
 }
+
+func (api *PublicDebugAPI) traceBlockByNumber(blockNum rpctypes.BlockNumber, config *evmtypes.TraceConfig) (interface{}, error) {
+
+	err := evmtypes.TestTracerConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("tracer err : %s", err.Error())
+	}
+	configBytes, err := json.Marshal(config)
```
Comment (Ywmet) [inline app/rpc/namespaces/debug/api.go:90]: always we use api.codec registry

**Review by Ywmet — DISMISSED** [review]

File: app/rpc/namespaces/debug/api.go (hunk 2 above)
Comment (lcmmhcc) [reply app/rpc/namespaces/debug/api.go:90]: it seems we often use json.marshal in rpc

File: app/rpc/namespaces/debug/api.go (hunk 1 above)
Comment (lcmmhcc) [reply app/rpc/namespaces/debug/api.go:26]: map[] not,  but its readonly

**Review by ItsFunny — APPROVED** [review]

**Review by Ywmet — APPROVED** [review]

**Pull request closed without merging**